src/mmap_unix.rs

```diff
@@ -74,6 +74,156 @@ impl error::Error for Error {}
 
 pub type Result<T> = result::Result<T, Error>;
 
+/// A factory struct to build `MmapRegion` objects.
+pub struct MmapRegionBuilder<B = ()> {
+    size: usize,
+    prot: i32,
+    flags: i32,
+    file_offset: Option<FileOffset>,
+    raw_ptr: Option<*mut u8>,
+    hugetlbfs: Option<bool>,
+    bitmap: B,
+}
+
+impl<B: Bitmap + NewBitmap> MmapRegionBuilder<B> {
+    /// Create a new MmapRegionBuilder.
+    pub fn new(size: usize) -> Self {
+        MmapRegionBuilder {
+            size,
+            prot: libc::PROT_READ | libc::PROT_WRITE,
+            flags: libc::MAP_ANONYMOUS | libc::MAP_NORESERVE | libc::MAP_PRIVATE,
+            file_offset: None,
+            raw_ptr: None,
+            hugetlbfs: None,
+            bitmap: B::with_len(size),
```

Comment:
I didn't review the dirty bitmap tracking PR, so sorry if this is a stupid question. Is it expected for all MmapRegions to have an associated bitmap? I was wondering if there is a way to opt out which we might want to factor in here in the builder.

I find it a bit strange that we initialize it here with a default value, and then we also have a `with_bitmap` function because that suggests that the Bitmap is not initialized beforehand.

Reply:
Changed bitmap to `Option<B>` and will create the required bitmap on demand.